include/alpaka/mem/Visibility.hpp

```diff
@@ -0,0 +1,128 @@
+/* Copyright 2023 Simeon Ehrig
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#include "alpaka/acc/Traits.hpp"
+#include "alpaka/dev/Traits.hpp"
+#include "alpaka/mem/view/ViewAccessOps.hpp"
+#include "alpaka/meta/ForEachType.hpp"
+#include "alpaka/meta/TypeListOps.hpp"
+#include "alpaka/platform/Traits.hpp"
+
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#define CREATE_MEM_VISIBILITY(mem_name)                                                                               \
+    struct mem_name                                                                                                   \
+    {                                                                                                                 \
+        static std::string get_name()                                                                                 \
+        {                                                                                                             \
+            return #mem_name;                                                                                         \
+        }                                                                                                             \
+    }
+
+namespace alpaka
+{
+    CREATE_MEM_VISIBILITY(MemVisibleCPU);
+    CREATE_MEM_VISIBILITY(MemVisibleFpgaSyclIntel);
+    CREATE_MEM_VISIBILITY(MemVisibleGenericSycl);
+    CREATE_MEM_VISIBILITY(MemVisibleGpuCudaRt);
+    CREATE_MEM_VISIBILITY(MemVisibleGpuHipRt);
+    CREATE_MEM_VISIBILITY(MemVisibleGpuSyclIntel);
+
```

Comment:
Can you `#undef CREATE_MEM_VISIBILITY` after using it ?
I assume users do not need to be able to use it directly.